podvm/Makefile

```diff
@@ -28,9 +22,14 @@ PACKER_DEFAULT_OPTS ?= -var boot_wait=300s
 
 image: $(IMAGE_FILE)
 
-setopts:
+setopts: tools
 ifeq ($(PODVM_DISTRO),ubuntu)
 	@echo defined
+
+	# Check for set but empty or unset URL and CHECKSUM
+	$(eval UBUNTU_IMAGE_URL ?= $(shell yq '.cloudimg.ubuntu.$(UBUNTU_RELEASE).$(DEB_ARCH).url' < ../versions.yaml))
+	$(eval UBUNTU_IMAGE_CHECKSUM ?= $(shell yq '.cloudimg.ubuntu.$(UBUNTU_RELEASE).$(DEB_ARCH).checksum' < ../versions.yaml))
+
```

Comment:
Can we refactor Makefile not to use the `setopts` target and the `eval` function  like this?  Variables defined with `=` are evaluated lazily. I don't think such refactoring is not the scope of this PR, but I think using `yq` for many variables makes Makefile more complicated, so I think it might be worth refactoring.

```

UBUNTU_IMAGE_URL = $(shell yq '.cloudimg.ubuntu.$(UBUNTU_RELEASE).$(DEB_ARCH).url' < ./versions.yaml)
UBUNTU_IMAGE_CHECKSUM = $(shell yq '.cloudimg.ubuntu.$(UBUNTU_RELEASE).$(DEB_ARCH).checksum' < ./versions.yaml)

ifeq ($(PODVM_DISTRO),ubuntu)

OPTS := -var qemu_image_name=$(IMAGE_FILE) \
	-var cloud_image_url=$(UBUNTU_IMAGE_URL) \
	-var cloud_image_checksum=$(UBUNTU_IMAGE_CHECKSUM)

ifneq ($(HOST_ARCH),$(ARCH))
OPTS += -var machine_type=$(QEMU_MACHINE_TYPE_$(ARCH))
ifndef QEMU_BINARY
OPTS += -var qemu_binary=qemu-system-$(ARCH)
endif
endif
endif

︙
$(IMAGE_FILE): $(BINARIES) $(FILES) tools
```

Reply:
I agree it is/could get messy, it is a larger task to refactor that which can be looked at in a different issue.